app/uk/gov/hmrc/gform/gform/FormController.scala

```diff
@@ -654,61 +662,57 @@ class FormController(
               val processData = purgeConfirmationData.f(processData0)
               val sn = maybePreviousSn.getOrElse(sectionNumber)
 
-              def goBack(saveData: Boolean) = {
-
+              def goBack() = {
                 val sectionTitle4Ga = formProcessor.getSectionTitle4Ga(processData, sn)
-                val fastForwardSn = routes.FormController
-                  .form(
-                    cache.formTemplateId,
-                    maybeAccessCode,
-                    sn,
-                    sectionTitle4Ga,
-                    SuppressErrors.Yes,
-                    fastForward
-                  )
-                val goBackLink = Redirect {
+
+                def backRoute(sectionNumber: SectionNumber, fastForward: FastForward) =
+                  routes.FormController
+                    .form(
+                      cache.formTemplateId,
+                      maybeAccessCode,
+                      sectionNumber,
+                      sectionTitle4Ga,
+                      SuppressErrors.Yes,
+                      fastForward
+                    )
+
+                val fastForwardSn: Call = {
+                  fastForward match {
+                    case FastForward.CYA(_, None) =>
+                      routes.SummaryController
+                        .summaryById(cache.formTemplateId, maybeAccessCode, commingFromSn.toCoordinates, None)
+                    case FastForward.CYA(_, Some(to)) =>
+                      backRoute(to, FastForward.Yes)
```

Comment:
if we use FastForward.Yes here we won't be able to go back several times and then got forward step by step